active/0000-blueprint-remove-old-files.md

```diff
@@ -0,0 +1,48 @@
+- Start Date: 2016-12-17
+- RFC PR: (leave this empty)
+- Ember CLI Issue: (leave this empty)
+
+# Summary
+
+Give blueprint generators the ability to clean up old files.
+
+# Motivation
+
+We want to eliminate the noise of having old files laying after updating
+ember-cli using `ember init`.
+
+# Detailed design
+
+We'd like a declarative API for blueprints to delete files instead of only
+create. It would be essentially syntactic sugar for removing the file yourself
+in an `afterInstall` hook. It would be an array on the blueprint's `index.js`.
+
+```js
+// ember-cli/bluprints/blah/index.js
+module.exports = {
+  // ...
+
+  oldFilesToRemove: [
```

Comment:
As this this implementation will not land before Jan 1, 2017. I believe we can use ES6 feature:

```
module.exports = {
  get existingFilesToRemove() {
    return [ /* ... */ ]
  }
}
```

This should enable both a reasonable API, and ensure the property exists on the instance and not the class.